### PR TITLE
user profiles CRUD

### PR DESCRIPTION
**Pull request by MicLi01**

### CODE REVIEW

File: server/controllers/profileHandler.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    });
+  }
+};
+
+const updateProfile = (req, res) => {
+  if(!req.isAuthenticated()){
+    res.status(401).json({error:"Must be logged in to perform this action"});
+  }
+  else{
+    userProfile.findOne({userId:req.user._id}, function(err, profile){
```
Comment (melvinheks) [inline server/controllers/profileHandler.js]: There's no need to find, use update

File: server/database/models/userProfile.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+
+const UserProfileSchema = new Schema({
+  summary:{type = String, default: ''},
+  userId:{type: Schema.Types.ObjectId, ref: 'User'}
```
Comment (melvinheks) [inline server/database/models/userProfile.js]: required

File: server/database/models/userProfile.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+
+const UserProfileSchema = new Schema({
+  summary:{type = String, default: ''},
```
Comment (melvinheks) [inline server/database/models/userProfile.js]: type:

File: server/database/models/userProfile.js (hunk 2 above)
Comment (MicLi01) [reply server/database/models/userProfile.js]: ok

File: server/database/models/userProfile.js (hunk 3 above)
Comment (MicLi01) [reply server/database/models/userProfile.js]: ok

File: server/controllers/profileHandler.js (hunk 1 above)
Comment (MicLi01) [reply server/controllers/profileHandler.js]: ok

**Comment by melvinheks** [comment]
POST
/api/account
* /createProfile
  * summary, img(url string)
* /retrieveProfile
  * [userId](if not supplied and logged in, retrieves profile of current logged in user)
* /updateProfile
  * summary, img(url string)
* /deleteProfile

**Pull request merged**